System/Reader.cpp

```diff
@@ -1,679 +1,448 @@
+						// Count the lines within the comment though
+						if (temp2 == '\n') { ++m_CurrentLine; }
+					}
+					// Eat that final '/'
+					if (!m_pStream->eof()) { m_pStream->ignore(1); }
+
+				// Not a comment, so it's data, so quit.
+				} else {
+					m_pStream->putback(temp);
+					break;
+				}
+			} else { 
+				break;
+			}
+		}
+
+		// Only do this if we actually ate an end line
+		// This precaution enables us to use Eat repeatedly without messing up the indentation tracking logic
+		if (ateLine) {
+			// Get indentation difference from the last line of the last call to Eat(),
+			// and the last line of this call to Eat().
+			m_IndentDifference = indent - m_PreviousIndent;
+			// Save the last tab count
+			m_PreviousIndent = indent;
+		}
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	std::string Reader::TrimString(std::string &stringToTrim) {
+		if (stringToTrim.empty()) { return ""; }
+
+		int start = stringToTrim.find_first_not_of(' ');
+		int end = stringToTrim.find_last_not_of(' ');
+
+		if (start > end) {
+			return "";
+		} else if (start == 0 && end == stringToTrim.size() - 1) {
+			return stringToTrim;
+		}
+		return stringToTrim.substr(start, (end - start) + 1);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Reader::ReportError(std::string errorDesc) {
+		char error[1024];
+		sprintf_s(error, sizeof(error), "%s Error happened in %s at line %i!", errorDesc.c_str(), m_FilePath.c_str(), m_CurrentLine);
+		RTEAbort(error);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool Reader::StartIncludeFile() {
+		// Report that we're including a file
+		if (m_fpReportProgress) {
+			char report[512];
+			sprintf_s(report, sizeof(report), "%s%s on line %i includes:", m_ReportTabs.c_str(), m_FileName.c_str(), m_CurrentLine);
+			m_fpReportProgress(string(report), false);
+		}
+		// Push the current stream onto the StreamStack for future retrieval when the new include file has run out of data.
+		m_StreamStack.push_back(StreamInfo(m_pStream, m_FilePath, m_CurrentLine, m_PreviousIndent));
+
+		// Get the file path from the stream
+		m_FilePath = ReadPropValue();
+		m_pStream = new std::ifstream(m_FilePath.c_str());
+		if (m_pStream->fail()) {
+			// Backpedal and set up to read the next property in the old stream
+			delete m_pStream;
+			m_pStream = m_StreamStack.back().m_pStream;
+			m_FilePath = m_StreamStack.back().m_FilePath;
+			m_CurrentLine = m_StreamStack.back().m_CurrentLine;
+			m_PreviousIndent = m_StreamStack.back().m_PreviousIndent;
+			m_StreamStack.pop_back();
+
+			// TODO: make this graceful log error instead.
+			ReportError("Failed to open included data file");
```

Comment:
Same here, `ReportError` already includes line and filepath.
Probably means not stopping the game, yeah, not sure how safe that is.

Reply:
If you wanna try it out without the ReportError go for it and we'll see how horribly it breaks. I don't really know what you could do if the includefile is screwed up, cancel loading the whole mod? It seems to me that exiting out would be the best solution even if the others didn't mean more work.

Reply:
Generally I think killing the game at any minor screwup is the best way to efficiently debug mods so I actually prefer no graceful logging anywhere.

Reply:
Fine with me. I guess the idea of doing this more gracefully would be a popup that says
```
X mod has failed to load for Y reason (whatever it would show you).
Do you want to skip this mod and continue to load the game?

(Yes) (No)
```
But that's a bunch of work for not a lot of benefit, so I don't think we should bother.
I'd say use your judgement on these and kill TODOs and commented code as needed.